deploy-tool/dns/gcloud/gcloud.go

```diff
@@ -0,0 +1,169 @@
+		return nil, err
+	}
+
+	svc, err := dns.NewService(context.Background(), option.WithHTTPClient(client))
+	if err != nil {
+		return nil, err
+	}
+
+	return &GoogleDNSProvider{
+		service:      svc,
+		project:      project,
+		zoneMappings: zoneMappings,
+	}, nil
+}
+
+func convertRecordType(record libdns.Record) dns.ResourceRecordSet {
+	return dns.ResourceRecordSet{
+		Name:    fmt.Sprintf("%s.", record.Name),
+		Rrdatas: []string{record.Value},
+		Ttl:     record.TTL.Nanoseconds() / 1e9, // Expected in seconds
+		Type:    record.Type,
+		Kind:    "dns#resourceRecordSet",
+	}
+}
+
+func (p *GoogleDNSProvider) convertZone(zone string) string {
+	newZone, ok := p.zoneMappings[zone]
+	if ok {
+		return newZone
+	}
+
+	return zone
+}
+
+// AppendRecords appends DNS records to a given zone
+func (p *GoogleDNSProvider) AppendRecords(ctx context.Context, zone string, recs []libdns.Record) ([]libdns.Record, error) {
+	zone = p.convertZone(zone)
+	gcpRecords := make([]*dns.ResourceRecordSet, len(recs))
+
+	for idx, record := range recs {
+		resourceRecordSet := convertRecordType(record)
+		gcpRecords[idx] = &resourceRecordSet
+	}
+	changes := &dns.Change{
+		Additions: gcpRecords,
+	}
+	result, err := p.service.Changes.Create(p.project, zone, changes).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	if result.Status == changeStatusDone {
+		return recs, nil
+	}
+
+	changeID := result.Id
+	err = Poll("append records", 30*time.Second, 3*time.Second, func() (bool, error) {
+		result, err := p.service.Changes.Get(p.project, zone, changeID).Do()
+		if err != nil {
+			return false, err
+		}
+
+		if result.Status == changeStatusDone {
+			return true, nil
+		}
+		return false, nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return recs, nil
+}
+
+// DeleteRecords deletes DNS records from a given zone
+func (p *GoogleDNSProvider) DeleteRecords(ctx context.Context, zone string, recs []libdns.Record) ([]libdns.Record, error) {
+	zone = p.convertZone(zone)
+	gcpRecords := make([]*dns.ResourceRecordSet, len(recs))
+
+	for idx, record := range recs {
+		resourceRecordSet := convertRecordType(record)
+		gcpRecords[idx] = &resourceRecordSet
+	}
+	changes := &dns.Change{
+		Deletions: gcpRecords,
+	}
+	result, err := p.service.Changes.Create(p.project, zone, changes).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	if result.Status == changeStatusDone {
+		return recs, nil
+	}
+
+	changeID := result.Id
+	err = Poll("delete records", 30*time.Second, 3*time.Second, func() (bool, error) {
+		result, err := p.service.Changes.Get(p.project, zone, changeID).Do()
+		if err != nil {
+			return false, err
+		}
+
+		if result.Status == changeStatusDone {
+			return true, nil
+		}
+		return false, nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return recs, nil
+}
+
+// Poll is a blocking polling for a given function with a given name. The poller must return true for the polling code to stop. The poller may also return an optional error to be reported
```

Comment:
Suggest wrapping this comment at 80 or 100 chars. Also does this function need to be exported from the module? If not, rename it `poll`.